### PR TITLE
Add stub brave vpn button to toolbar

### PR DESCRIPTION
**Pull request by simonhong**
fix brave/brave-browser#17235

Note: BraveVPN feature is disabled by default.
To enable, pass `--enable-features=BraveVPN` command line options.

Screenshot 2021-08-03 110025
Screenshot 2021-08-03 110051
Screenshot 2021-08-04 010159
Screenshot 2021-08-03 234147

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: browser/ui/views/toolbar/brave_vpn_button.cc [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  std::unique_ptr<views::Border> border = views::CreateRoundedRectBorder(
+      1, kHighlightRadius, gfx::Insets(), kBorderColor);
+
+  constexpr gfx::Insets kTargetInsets{4, 6};
+  const gfx::Insets extra_insets = kTargetInsets - border->GetInsets();
+  SetBorder(views::CreatePaddedBorder(std::move(border), extra_insets));
+
+  constexpr int kBraveAvatarImageLabelSpacing = 4;
+  SetImageLabelSpacing(kBraveAvatarImageLabelSpacing);
+  return;
```
Comment (keur) [inline browser/ui/views/toolbar/brave_vpn_button.cc:92]: nit: no-op

File: browser/ui/views/toolbar/brave_vpn_button.cc [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+  label()->SetFontList(
+      font_list.DeriveWithSizeDelta(kFontSize - font_list.GetFontSize()));
+
+  // Set image positions first. then label.
+  SetHorizontalAlignment(gfx::ALIGN_LEFT);
+
+  UpdateButtonState();
+}
+
+BraveVPNButton::~BraveVPNButton() = default;
```
Comment (keur) [inline browser/ui/views/toolbar/brave_vpn_button.cc:69]: nit: can put this in header

File: browser/ui/views/toolbar/brave_vpn_button.cc [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+  NOTIMPLEMENTED();
+  return true;
+}
+
+void BraveVPNButton::OnButtonPressed(const ui::Event& event) {
+  ShowBraveVPNPanel();
+}
+
+void BraveVPNButton::ShowBraveVPNPanel() {
+  NOTIMPLEMENTED();
```
Comment (keur) [inline browser/ui/views/toolbar/brave_vpn_button.cc:119]: missing include for `base/notreached.h`?

File: browser/ui/views/toolbar/brave_vpn_button.cc [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+#include "ui/gfx/geometry/size.h"
+#include "ui/gfx/paint_vector_icon.h"
+#include "ui/gfx/skia_util.h"
+#include "ui/views/background.h"
+#include "ui/views/border.h"
+#include "ui/views/controls/highlight_path_generator.h"
+
+namespace {
+
+constexpr int kHighlightRadius = 47;
```
Comment (keur) [inline browser/ui/views/toolbar/brave_vpn_button.cc:27]: is this the thickness in pixels of highlight? comment here would be nice

File: components/brave_vpn/brave_vpn_utils.h [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_VPN_BRAVE_VPN_UTILS_H_
+#define BRAVE_COMPONENTS_BRAVE_VPN_BRAVE_VPN_UTILS_H_
+
+namespace brave_vpn {
+
+bool IsBraveVPNEnabled();
+
+}
```
Comment (keur) [inline components/brave_vpn/brave_vpn_utils.h:13]: nit: missing terminating namespace comment

**Review by karenkliu — CHANGES_REQUESTED** [review]
Hi! Sorry - I just added the dark theme colors. Please take a look and tweak the VPN button style for dark theme: https://www.figma.com/file/qtMYwoj3JWJO5xzwLt6j2f/?node-id=0%3A1

image

**Review by simonhong — COMMENTED** [review]
@karenkliu dark theme updated. PTAL again.

**Review by keur — APPROVED** [review]
LGTM

**Review by karenkliu — CHANGES_REQUESTED** [review]
Hi @simonhong , when VPN is disconnected it should actually look like this:

Screen Shot 2021-08-03 at 7 12 47 PM

I decided not to have the "Disconnected" text to avoid taking up too much space in the toolbar!

**Review by karenkliu — APPROVED** [review]
Looks great! 🚀

**Comment by simonhong** [comment]
Merged because previous CI run got all green and just rebased on master with trivial conflict.

**Pull request merged**